addons/crm/static/src/js/fields/many2one_avatar_leader_user.js

```diff
@@ -0,0 +1,103 @@
+import { Many2XAvatarUserAutocomplete } from "@mail/views/web/fields/avatar_autocomplete/avatar_many2x_autocomplete";
+import {
+    Many2OneAvatarUser,
+    many2OneAvatarUserField,
+    Many2OneAvatarUserField,
+} from "@mail/views/web/fields/many2one_avatar_user_field/many2one_avatar_user_field";
+import { onWillUpdateProps, useState } from "@odoo/owl";
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+
+class Many2XAvatarUserLeaderAutocomplete extends Many2XAvatarUserAutocomplete {
+    static props = {
+        ...Many2XAvatarUserAutocomplete.props,
+        leaderState: Object,
+    };
+
+    /**
+     * @override
+     */
+    buildRecordSuggestion(request, record) {
+        if (record.id == this.props.leaderState.teamLeaderId) {
```

Comment:
Would say it is not the role of JS to do that, especially we introduced formatting based on ``formatted_display_name`` context key. Can't we propagate a context key ``crm_formatted_display_name`` used in some places, and override display_name of user model to add odoomarkup + Team Leader string in python, based on that context key ? Would be more standard, considering how the odoomark was implemented.